Source/Core/DolphinQt/Config/GameConfigHighlighter.cpp

```diff
@@ -65,17 +69,42 @@ GameConfigHighlighter::GameConfigHighlighter(QTextDocument* parent) : QSyntaxHig
       HighlightingRule{QRegularExpression(QStringLiteral("^\\$.*")), comment_format});
   m_rules.emplace_back(
       HighlightingRule{QRegularExpression(QStringLiteral("^\\*.*")), comment_format});
+
+  // Highlight block on change.
+  // We're manually highlighting blocks because QSyntaxHighlighter
+  // hangs with large (>2MB) files for some reason.
+  connect(parent, &QTextDocument::contentsChange, this,
+          [this, parent](int pos, int removed [[maybe_unused]], int added) {
+            QTextBlock block = parent->findBlock(pos);
+            const auto pos_end = pos + added;
+            while (block.isValid() && block.position() <= pos_end)
+            {
+              HighlightBlock(block);
+              block = block.next();
+            }
+          });
+
+  // Highlight all blocks right now.
+  for (QTextBlock block = parent->begin(); block.isValid(); block = block.next())
+    HighlightBlock(block);
```

Comment:
You removed the default nullptr parameter so GameConfigHighlighter won't be default contructed; are we assuming nobody's going to pass nullptr in explicitly?

Reply:
Yeah. The parent is also the `QTextDocument` to be formatted. There shouldn't be a need to create a `GameConfigHighlighter` if there's no document.